x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1223,8 +1222,7 @@ private Expression resolveConvertFunction(AbstractConvertFunction convert, List<
                 HashMap<TypeResolutionKey, Expression> typeResolutions = new HashMap<>();
                 Set<DataType> supportedTypes = convert.supportedTypes();
                 imf.types().forEach(type -> {
-                    // TODO: Shouldn't we perform widening of small numerical types here?
-                    if (supportedTypes.contains(type)) {
+                    if (supportedTypes.contains(type.widenSmallNumeric())) {
```

Comment:
Contrary to the funny `TODO` that I left here, haha, maybe we should actually widen at the point where the `InvalidMappedField` gets created? That'd be consistent with how we widen the types of regular `EsField`s in `Analyzer.mappingAsAttributes`. If we do so, we could enforce this as an invariant of `FieldAttribute` (so that when it contains an IMF, it has to be widened).